### ISSUE TITLE
configuration: linking to an Xcode project produce the wrong paths

### ISSUE THREAD

**Issue by DivineDominion** [krzysztofzablocki/Sourcery#619]
When I want to have results of templates added to my Xcode project, I _do_ get the group and file entries with the correct name, and as expected both are "Relative to Group"; but the file reference doesn't resolve since it always has a `../` too many.

If the group's name doesn't match the output path's folder name, though, the group will point to the wrong directory, too. It's a group _with_ folder reference in Xcode, not without.

The group setting looks okay, provided the `EntityInfoTest-generated/` folder exists!

![Xcode files](https://user-images.githubusercontent.com/59080/40265825-036f6980-5b40-11e8-806a-c71d8476476e.png)  
![group settings](https://user-images.githubusercontent.com/59080/40265801-840b0cbc-5b3f-11e8-95f6-6b03b38f6eda.png)

But the file reference of the generated file is off:

![generated file](https://user-images.githubusercontent.com/59080/40265807-9c4863a6-5b3f-11e8-80f0-1a2a59d4f11d.png)

Instead of `../../EntityInfo.generated.swift`, it should just be `EntityInfo.generated.swift`.

So the issues I see are:

- The Xcode group will only point to the correct folder if output.path and output.link.group have to have the same value
- File references relative to the group shall have a relative path of `./`, not `../../` 

## Setup 

- I checked out sourcery as a project local submodule into extern/Sourcery and built from there
- I run the CLI program via `./external/Sourcery/.build/debug/sourcery`
- The config file (`.sourcery.yml`) looks as follows:

```yml
project:
  file: Foo.xcodeproj
  target:
    name: Foo
    module: Foo
templates:
  - ./templates/
output: 
  path: ./Foo-generated/
  link:
    project: Foo.xcodeproj
    target: Foo
    group: Foo-generated // if this is not == path, e.g. "Sourcery", you'll be in trouble, too
```

---

Xcode 9.3
Apple Swift version 4.1 (swiftlang-902.0.48 clang-902.0.37.1)
Target: x86_64-apple-darwin17.5.0

**Comment by ilyapuchka** [comment]
Can you upload a sample project that demonstrates this issue?

**Comment by DivineDominion** [comment]
Here's a sample project: https://github.com/DivineDominion/SourceryTestApp

Run `rake` from the CLI to checkout the submodule, build, and then run Sourcery.

The configuration:

```yml
project:
  file: SourceryTestApp.xcodeproj
  target:
    name: SourceryTestApp
    module: SourceryTestApp
templates:
  - ./templates/
output:
  path: ./SourceryTestApp-generated/
  link:
    project: SourceryTestApp.xcodeproj
    target: SourceryTestApp
    group: SourceryTestApp-generated
```

Interestingly, I notice an error in the console:

```
$ rake run
Using configuration file at '.sourcery.yml'
Scanning sources...
warning: Skipping file at /Users/ctm/Downloads/SourceryTestApp-generated/Test.generated.swift as it does not exist
Found 0 types.
Loading templates...
Loaded 1 templates.
Generating code...
Finished.
Processing time 1.04168808460236 seconds
```

- `/Users/ctm/Downloads/SourceryTestApp/` is the project path, and Sourcery reports 
- `/Users/ctm/Downloads/SourceryTestApp-generated/Test.generated.swift` to be not found instead of the expected 
- `/Users/ctm/Downloads/SourceryTestApp/SourceryTestApp-generated/Test.generated.swift`

**Comment by ilyapuchka** [comment]
Thanks, @DivineDominion, this helped to spot the issue, should be fixed on master now.

**Comment by DivineDominion** [comment]
Not quite, yet. With these settings:

```
output:
  path: ./MyCustomPath/
  link:
    project: Foo.xcodeproj
    target: Foo
    // group: Foo-generated // if this is not == path, e.g. "Sourcery", you'll be in trouble, too
```

... the code will be generated into `MyCustomPath/` but the Xcode group will reference a folder called `SourceryGenerated/` (that does not exist).

**Comment by ilyapuchka** [comment]
This should be fixed with 1f79d695dd196b508b90b5cd358f74cb09f33fb8
`path` is always required, `group` if missing will now add files to root group, as described in documentation, and folders will be created for any new group added.
